src/ring_theory/algebra.lean

```diff
@@ -723,33 +723,109 @@ section restrict_scalars
 `S`-modules are also `R`-modules. -/
 
 variables (R : Type*) [comm_ring R] (S : Type*) [ring S] [algebra R S]
-(E : Type*) [add_comm_group E] [module S E] {F : Type*} [add_comm_group F] [module S F]
+variables (E : Type*) [add_comm_group E] [module S E] {F : Type*} [add_comm_group F] [module S F]
 
-/-- When `E` is a module over a ring `S`, and `S` is an algebra over `R`, then `E` inherits a
-module structure over `R`, called `module.restrict S R E`.
-Not registered as an instance as `S` can not be inferred. -/
-def module.restrict_scalars : module R E :=
-{ smul      := λc x, (algebra_map R S c) • x,
+/--
+When `E` is a module over a ring `S`, and `S` is an algebra over `R`, then `E` inherits a
+module structure over `R`, called `module.restrict_scalars' R S E`.
+We do not register this as an instance as `S` can not be inferred.
+-/
+def module.restrict_scalars' : module R E :=
+{ smul      := λ c x, (algebra_map R S c) • x,
   one_smul  := by simp,
   mul_smul  := by simp [mul_smul],
   smul_add  := by simp [smul_add],
   smul_zero := by simp [smul_zero],
   add_smul  := by simp [add_smul],
   zero_smul := by simp [zero_smul] }
 
+/--
+When `E` is a module over a ring `S`, and `S` is an algebra over `R`, then `E` inherits a
+module structure over `R`, provided as a type synonym `module.restrict_scalars R S E := E`.
+-/
+@[nolint unused_arguments]
+def module.restrict_scalars (R : Type*) (S : Type*) (E : Type*) : Type* := E
```

Comment:
In the current version, `module.restrict_scalars` does not include the typeclass assumptions on `R`, `S` and `E` (and the `unused_argument` is useless). Could you add them explicitly?

Reply:
Done!